pkg/reconciler/buildrun/resources/taskrun.go

```diff
@@ -165,6 +173,16 @@ func GenerateTaskSpec(
 
 		taskImage := getStringTransformations(containerValue.Image)
 
+		if containerValue.Env == nil {
+			containerValue.Env = []corev1.EnvVar{}
+		}
+
+		// Any collision between the env vars in the Container step and those in the Build/BuildRun
+		// will result in an error and cause a failed TaskRun
+		if containerValue.Env, err = env.MergeEnvVars(combinedEnvs, containerValue.Env, false); err != nil {
+			return &generatedTaskSpec, fmt.Errorf("error(s) occurred merging environment variables into BuildStrategy %q steps: %s", build.Spec.StrategyName(), err.Error())
+		}
```

Comment:
Ähm, following suggestion which I do not know if it is necessary ... we atm have a reconciler for (cluster) build strategies. We actually do not do anything there but that's a different story. But, a side effect is that the k8s client imo caches these objects. I am very careful in mutating objects that I get from a cache and that's what you are potentially doing here. Now, I am pretty sure that k8s client copies objects so that this would not be a problem, and potentially some object in our call reference might be passed by value anyway. Anyway, assumptions vs potential changes in the future, and if it ever does we have an ugly bug to find. So, in short, while I am 99.99 % certain that your code works, I would feel more confident like this:

```suggestion
		stepEnv := containerValue.Env
		if stepEnv == nil {
			stepEnv = []corev1.EnvVar{}
		}

		// Any collision between the env vars in the Container step and those in the Build/BuildRun
		// will result in an error and cause a failed TaskRun
		if stepEnv, err = env.MergeEnvVars(combinedEnvs, stepEnv, false); err != nil {
			return &generatedTaskSpec, fmt.Errorf("error(s) occurred merging environment variables into BuildStrategy %q steps: %s", build.Spec.StrategyName(), err.Error())
		}
```

Line 197 then need to be changed to `Env: stepEnv`